### PR TITLE
[kube-prometheus-stack] Exclude default rules from namespace enforcement

### PR DESCRIPTION
**Pull request by amuraru**
#### What this PR does / why we need it:

When enforcedNamespaceLabel is enabled the default rules installed by the
helm chart also also changed dynamically by the operator at runtime.
This breaks cluster wide recorded rules and alert rules.

This patch ensures that default prometheus rules are automatically
added to `prometheusRulesExcludedFromEnforce`

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by amuraru** [comment]
also see https://github.com/prometheus-operator/prometheus-operator/issues/3919

**Comment by Xtigyro** [comment]
Hey @amuraru 👋 
Thank you for the contribution - it's appreciated! 🥇 

I'll be having a look at this PR today and hopefully will be merged by Monday.

File: charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml [hunk 1]
```diff
@@ -292,10 +289,19 @@ spec:
 {{- if .Values.prometheus.prometheusSpec.ignoreNamespaceSelectors }}
   ignoreNamespaceSelectors: {{ .Values.prometheus.prometheusSpec.ignoreNamespaceSelectors }}
 {{- end }}
-{{- if .Values.prometheus.prometheusSpec.prometheusRulesExcludedFromEnforce }}
+{{- if .Values.prometheus.prometheusSpec.enforcedNamespaceLabel }}
```
Comment (amuraru) [inline charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml:292]: I moved this here and nest with the `prometheusDefaultRulesExcludedFromEnforce` and `prometheusRulesExcludedFromEnforce` as the latters are not applicable unless `enforcedNamespaceLabel` is set

**Review by Xtigyro — CHANGES_REQUESTED** [review]
@amuraru Could we please think of a way for automating the maintenance of that explicit list? With this PR if you add a new group of rules - you'll need to add it to one more place.

**Comment by amuraru** [comment]
> Could we please think of a way for automating the maintenance of that explicit list? With this PR if you add a new group of rules - you'll need to add it to one more place.

thanks for your suggestion @Xtigyro - please check the new version, the patch now generates automatically the rules names dynamically in sync_prometheus_rules.py and stores them as a yaml file where they are dynamically loaded from.
lmk

**Comment by Xtigyro** [comment]
@amuraru Very nice! And how about naming `files` - `aux`?

**Comment by amuraru** [comment]
What about `generated` and try to move generated rules file there as well?

**Comment by amuraru** [comment]
@Xtigyro please check again - I changed the generated yaml file to a helm named template.
I think it looks more compact now

**Review by Xtigyro — APPROVED** [review]

**Comment by Xtigyro** [comment]
@amuraru Sweet! Many thanks for your excellent work! 💯

**Pull request merged**